File: Leetcode977.java
public class Leetcode977 {
    public int[] sortedSquares(int[] nums) {
        int left = 0, right = nums.length-1;
        // 뒤에서부터 채워 나가는 이유 -> 제곱값은 무조건 양수, 최대값을 찾아서 뒤에서부터 채워 나가야 함
        int idx = nums.length-1;
        int[] answer = new int[nums.length];
        while(left <= right){
            int lvalue = nums[left]*nums[left];
            int rvalue = nums[right]*nums[right];

            if(lvalue>=rvalue){
                answer[idx] = lvalue;
                left += 1;
            }
            else{
                answer[idx] = rvalue;
                right -= 1;
            }
            idx -= 1;
        }
        return answer;
    }
}

File: Python/Leetcode977.py
class Leetcode977:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        start = 0; end = len(nums)-1
        idx = len(nums)-1
        answer = [0 for i in range(len(nums))]

        while(start<=end):
            if nums[start]**2 >= nums[end]**2:
                answer[idx] = nums[start]**2
                start += 1
                idx -= 1
            else:
                answer[idx] = nums[end]**2
                end -= 1
                idx -= 1

        return answer